lib/compress/huf_compress.c

```diff
@@ -1244,6 +1235,63 @@ typedef struct {
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_SIZE 4096
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_RATIO 10  /* Must be >= 2 */
 
+unsigned HUF_cardinality(const unsigned* count, unsigned maxSymbolValue)
+{
+    unsigned cardinality = 0;
+    unsigned i;
+
+    for (i = 0; i < maxSymbolValue + 1; i++) {
+        if (count[i] != 0) cardinality += 1;
+    }
+
+    return cardinality;
+}
+
+unsigned HUF_minTableLog(size_t srcSize, unsigned symbolCardinality)
+{
+    U32 minBitsSrc = ZSTD_highbit32((U32)(srcSize)) + 1;
```

Comment:
I would simplify this function.

The reason `srcSize` was previously part of the formula
is that `maxSymbolValue` used to be a generous upper bound of the real cardinality.
Using `srcSize` as a second estimator would impact cases where `srcSize < maxSymbolValue`,
and producing a tighter upper bound estimate in these cases.

But now that we have the real cardinality, there is no need for another approximative upper bound.
Derive the result from `symbolCardinality` directly.

Reply:
This makes sense thanks for explaining, I initially wasn't sure why we did the first check.